### PR TITLE
Fix/remove rapid

### LINKED ISSUE #16: Remove RAPID
## Use Case
*Please provide a use case to help us understand your request in context*

## Solution
*Please describe your ideal solution*

## Alternatives
*Please describe any alternatives you've considered, even if you've dismissed them*

### PR DESCRIPTION
**Pull request by meganrm**
Problem
=======
closes #16 

Solution
========
Removed all RAPID imports, files and functions

## Type of change
Please delete options that are not relevant.

* Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: cellpack/autopack/ingredient/grow.py [hunk 1]
```diff
@@ -7,7 +7,6 @@
 from panda3d.core import Point3, TransformState
 from panda3d.bullet import BulletSphereShape, BulletRigidBodyNode
 from random import uniform, gauss, random
-from time import time
```
Comment (mogres) [inline cellpack/autopack/ingredient/grow.py:10]: Is this related to rapid or just an unnecessary import?

**Review by mogres — APPROVED** [review]
Looks good to me! This was an interesting exploration into the code.
I searched "rapid" to look for possible missed replacements and couldn't find anything.

File: cellpack/autopack/BaseGrid.py [hunk 2]
```diff
@@ -886,74 +885,6 @@ def computeVolume(self, space=None, fbox_bb=None):
...
-            numpy.identity(3),
-            numpy.array([0.0, 0.0, 0.0], "f"),
-            1.0,
-            ray_model,
-            RAPIDlib.cvar.RAPID_ALL_CONTACTS,
-        )
-        # could display it ?
-        return RAPIDlib.cvar.RAPID_num_contacts
-
-    def checkPointInside_rapid(self, point, diag, ray=1):
```
Comment (corredD) [inline cellpack/autopack/BaseGrid.py:935]: this function is called in getSurfaceInnerPoints_jordan so it needs to have an alternative or completely remove getSurfaceInnerPoints_jordan.

File: cellpack/autopack/Compartment.py [hunk 3]
```diff
@@ -1096,44 +1074,6 @@ def createSurfacePoints(self, maxl=20):
...
-            numpy.identity(3),
-            numpy.array([0.0, 0.0, 0.0], "f"),
-            1.0,
-            ray_model,
-            RAPIDlib.cvar.RAPID_ALL_CONTACTS,
-        )
-        # could display it ?
-        return RAPIDlib.cvar.RAPID_num_contacts
-
     def checkPointInside_rapid(self, point, diag, ray=1):
```
Comment (corredD) [inline cellpack/autopack/Compartment.py:1077]: shouldnt this function be removed ?

File: cellpack/autopack/ingredient/grow.py (hunk 1 above)
Comment (corredD) [reply cellpack/autopack/ingredient/grow.py:10]: not related to rapid, used in development for timing the execution of the growing algorithm.

**Review by corredD — COMMENTED** [review]
The overall removal of Rapid raise the issue of testing with compartment and that updated approach to voxelize the compartment meshes are needed. An interesting approach can be found here : https://github.com/NVIDIAGameWorks/FleX/blob/master/core/sdf.cpp . Another one with gpu can be found here : https://github.com/Forceflow/cuda_voxelizer and https://code.google.com/archive/p/voxelpipe/.

File: cellpack/autopack/ingredient/Ingredient.py [hunk 4]
```diff
@@ -3043,20 +3015,6 @@ def attempt_to_pack_at_grid_location(
                 moving,
```
Comment (mogres) [inline cellpack/autopack/ingredient/Ingredient.py:3015]: cellpack/autopack/ingredient/Ingredient.py Line 2546 `def get_rapid_nodes(self, close_indice, curentpt, removelast=False, prevpoint=None):`

Is this function still required after removing rapid?

PS: for some reason github won't let me leave comments at line 2546, so leaving it here for now.

File: cellpack/autopack/Compartment.py (hunk 3 above)
Comment (mogres) [reply cellpack/autopack/Compartment.py:1077]: It is referenced a few times:
1) cellpack/autopack/BaseGrid.py Line 408 `inside = self.checkPointInside_rapid(`
2) cellpack/autopack/BaseGrid.py Line 939 `r = self.checkPointInside_rapid(coord, diag, ray=ray)`
3) cellpack/autopack/Compartment.py Line 1856 `inside = self.checkPointInside_rapid(coord, diag, ray=3)`
4) cellpack/autopack/Compartment.py Line 1875 `inside = self.checkPointInside_rapid(coord, diag, ray=3)`
5) cellpack/autopack/Compartment.py Line 3832 `r = self.checkPointInside_rapid(coord, diag, ray=ray)`
6) cellpack/autopack/ingredient/Ingredient.py Line 2191 `inside = organelle.checkPointInside_rapid(point, self.env.grid.diag, ray=3)`
7) cellpack/autopack/ingredient/Ingredient.py Line 2202 `inside = o.checkPointInside_rapid(point, self.env.grid.diag, ray=3)`

These should probably also be removed if this function is going to be removed.

**Comment by mogres** [comment]
RECIPE_SCHEMA.md should probably also be updated at some point.

File: cellpack/autopack/ingredient/Ingredient.py (hunk 4 above)
Comment (meganrm) [reply cellpack/autopack/ingredient/Ingredient.py:2927]: thanks, removed!

File: cellpack/autopack/Compartment.py (hunk 3 above)
Comment (meganrm) [reply cellpack/autopack/Compartment.py]: yes, I removed this and replaced calls to it with `checkPointInside`
